File: examples/javascript/bh1749.js
/*
* The MIT License (MIT)
*
* Author: Assam Boudjelthia
* Copyright (c) 2018 Rohm Semiconductor.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var bh1749 = require("jsupm_bh1749");

var bh1749_sensor = new bh1749.BH1749();

bh1749_sensor.SensorInit(
	bh1749.INT_JUDGE_1,
	bh1749.MEAS_240MS,
	bh1749.RGB_GAIN_1X,
	bh1749.IR_GAIN_1X,
	bh1749.RED);
bh1749_sensor.SetThresholdHigh(511);
bh1749_sensor.Enable();

var waitTime = bh1749_sensor.GetMeasurementTime();
var counter = 10;
console.log("Color readings: ");
var interval = setInterval(function() {
	data = bh1749_sensor.GetMeasurements();
	console.log(
		"R: " + data.get(0) + ", G: " + data.get(1) + ", B: " + data.get(2) +
		", IR: " + data.get(3) + ", G2: " + data.get(4)
	);
	counter--;
	if (counter == 0) {
		clearInterval(interval);
	}
}, waitTime);

process.on("SIGINT", function() {
	clearInterval(interval);
	console.log("Exiting...");
	process.exit(0);
});

File: examples/c++/bh1749.cxx
/*
* The MIT License (MIT)
*
* Author: Assam Boudjelthia
* Copyright (c) 2018 Rohm Semiconductor.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

#include <iostream>
#include <unistd.h>
#include <stdexcept>
#include <signal.h>
#include <vector>
#include "bh1749.hpp"
#include "upm_utilities.h"

bool isStopped;
upm::BH1749 dev;

void signal_int_handler(int signo)
{
    if (signo == SIGINT)
        isStopped = true;
}

//! [Interesting]
void PrintData(void *args)
{
    std::vector<uint16_t> result = dev.GetMeasurements();
    std::cout << "R: " << result.at(0) <<
                ", G: " << result.at(1) <<
                ", B: " << result.at(2) <<
                ", IR: " << result.at(3) <<
                ", G2: " << result.at(4) << std::endl;
    dev.ResetInterrupt();
}

int main(int argc, char **argv)
{
    signal(SIGINT, signal_int_handler);
    dev.SoftReset();
    dev.SensorInit(INT_JUDGE_1, MEAS_240MS, RGB_GAIN_1X, IR_GAIN_1X, RED);
    dev.SetThresholdHigh(511);
    std::cout << "Installing ISR" << std::endl;
    dev.InstallISR(MRAA_GPIO_EDGE_FALLING, 33, &PrintData, NULL);
    dev.EnableInterrupt();
    dev.Enable();

    while(!isStopped) {
        upm_delay_ms(1000);
    }

    return 0;
}
//! [Interesting]
